libtenzir/src/healthmetrics_collector.cpp

```diff
@@ -0,0 +1,57 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/data.hpp>
+#include <tenzir/detail/weak_run_delayed.hpp>
+#include <tenzir/healthmetrics_collector.hpp>
+#include <tenzir/import_stream.hpp>
+#include <tenzir/series_builder.hpp>
+
+namespace tenzir {
+
+auto healthmetrics_collector(
+  healthmetrics_collector_actor::stateful_pointer<healthmetrics_collector_state>
+    self,
+  caf::timespan collection_interval, const node_actor& node)
+  -> healthmetrics_collector_actor::behavior_type {
+  // Build and import metric event.
+  auto stream = import_stream::make(self, node);
+  TENZIR_ASSERT_CHEAP(stream); // FIXME error handling
+  self->state.importer = std::make_unique<import_stream>(std::move(*stream));
+  self->state.collection_interval = collection_interval;
+  for (auto const* plugin : plugins::get<health_metrics_plugin>()) {
+    auto name = plugin->name();
+    auto collector = plugin->make_collector();
+    self->state.collectors[name] = collector;
+  }
+  return {
+    [self](atom::run) -> caf::result<void> {
```

Comment:
I think is suffices to start this automatically with the actor itself. One change I would make here is to run the collection once immediately before entering the loop.